### PR TITLE
Add insecure option to requests

### PR DESCRIPTION
**Pull request by mariusbertram**
I fixed the found Bug in #95 .
Now all requests have the option to skip tls verification.

### CODE REVIEW

File: oras/provider.py [hunk 1]
```diff
@@ -348,7 +353,7 @@ def get_blob(
         """
         method = "GET" if not head else "HEAD"
         blob_url = f"{self.prefix}://{container.get_blob_url(digest)}"  # type: ignore
-        return self.do_request(blob_url, method, headers=self.headers, stream=stream)
+        return self.do_request(blob_url, method, headers=self.headers, stream=stream, insecure=self._insecure )
```
Comment (vsoch) [inline oras/provider.py:356]: Thank you for opening this! 

Perhaps we don't need to pass self._insecure into a function that already is on the class - we could just use it in the function without doing that?

**Review by vsoch — COMMENTED** [review]
Thank you for opening this! Will you have any issues signing the DCO? This is a CNCF project so it's standard policy.

File: oras/provider.py (hunk 1 above)
Comment (mariusbertram) [reply oras/provider.py]: You are right. I made a new commit where the function do_request gets the insecure parameter from the class

File: oras/provider.py [hunk 2]
```diff
@@ -348,7 +352,7 @@ def get_blob(
         """
         method = "GET" if not head else "HEAD"
         blob_url = f"{self.prefix}://{container.get_blob_url(digest)}"  # type: ignore
-        return self.do_request(blob_url, method, headers=self.headers, stream=stream)
+        return self.do_request(blob_url, method, headers=self.headers, stream=stream )
```
Comment (vsoch) [inline oras/provider.py:355]:
```suggestion
        return self.do_request(blob_url, method, headers=self.headers, stream=stream)
```

File: oras/provider.py [hunk 3]
```diff
@@ -842,6 +846,7 @@ def do_request(
             json=json,
             headers=headers,
             stream=stream,
+            verify=not self._insecure
```
Comment (vsoch) [inline oras/provider.py:849]: Perfecto! This is exactly what I had in mind.

File: oras/provider.py (hunk 2 above)
Comment (vsoch) [reply oras/provider.py:355]: This (and the trailing comma you edited) will likely fail the formatting check. I will approve this to run but we will still need the DCO signed (let me know if you have questions about that).

Comment (mariusbertram) [reply oras/provider.py:355]: I signed the commits with my ssh key. Or does it needed with with gpg ? I will fix the formarting.

Comment (mariusbertram) [reply oras/provider.py]: Formarting is back to origin :D

Comment (vsoch) [reply oras/provider.py]: It's something called a DCO that you need to sign for CNCF projects - click the link in the failed step below, e.g., https://github.com/oras-project/oras-py/pull/96/checks?check_run_id=15068749650

Comment (mariusbertram) [reply oras/provider.py]: Ahh I didn't know that. Its done

**Comment by vsoch** [comment]
ah - there was a bump to black that introduced some error. I can fix this for you.

**Pull request merged**